### PR TITLE
HW-11: Typescript foundations

### PR DESCRIPTION
**Pull request by lenanova**

### CODE REVIEW

**Comment by lenanova** [comment]
Hi @anatoly-karpovich @VladimirLQA Please help to review this PR.

File: hw11/task2.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    address: {
+        country: "USA",
+        city: "San Francisco",
+        street: "El Camino Real",
+        house: 100,
+        flat: 19,
+    }
+}
+
+function isItEmployee (employee: IEmployee | ItEmployee): employee is ItEmployee {
```
Comment (anatoly-karpovich) [inline hw11/task2.ts:35]: awesome! I knew you can solve this!

File: hw11/task2.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            const value = obj[key];
+            if (typeof value === "string") {
+                result.string += 1;
+            } else if (typeof value === "number") {
+                result.number += 1;
+            } else if (typeof value === "boolean") {
+                result.boolean += 1;
+            }
+        }
+    }
```
Comment (anatoly-karpovich) [inline hw11/task2.ts]: Better use this count function as an ouside helper, and not being connected to "parent" function's result variable
```suggestion
    const countValues = (obj: TObject) => {
        const result: TCount = { string: 0, number: 0, boolean: 0 }
        for (const key in obj) {
            const value = obj[key];
            if (typeof value === "string") {
                result.string += 1;
            } else if (typeof value === "number") {
                result.number += 1;
            } else if (typeof value === "boolean") {
                result.boolean += 1;
            }
        }
        return result
    }
```

File: hw11/task2.ts [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    filter(numbers, (n) => n > 3); // [4, 133]
+    filter(numbers, (n) => n % 2 == 0); // [2, 4]
+    Параметры функции: Массив чисел и Анонимная функция, принимающая на вход число 
+    и возвращающая логическое значение.
+*/
+
+type TcallbackFunction = (n: number) => boolean;
+type Tnumbers = number[];
+
+function filter(numbers: Tnumbers, callbackFunction: TcallbackFunction): Tnumbers {
```
Comment (anatoly-karpovich) [inline hw11/task2.ts:141]:
```suggestion
function filter(numbers: number[], callbackFunction: TcallbackFunction): Tnumbers {
```
No need to create extra type for such simple argument

**Review by anatoly-karpovich — APPROVED** [review]
Well Done!

**Review by VladimirLQA — APPROVED** [review]

**Pull request merged**